Jenkinsfile

```diff
@@ -43,389 +53,412 @@ pipeline {
 
   agent any
   stages {
-    stage ('Stop same job builds') {
+    stage ('Pre-build') {
       agent { label 'master' }
       steps {
         script {
+          load ".jenkinsci/enums.groovy"
+          def preBuildRoutine = load ".jenkinsci/pre-build.groovy"
+          preBuildRoutine.prepare()
           if (GIT_LOCAL_BRANCH != "develop") {
             def builds = load ".jenkinsci/cancel-builds-same-job.groovy"
             builds.cancelSameJobBuilds()
           }
         }
       }
     }
-    stage('Build Debug') {
-      when {
-        allOf {
-          expression { params.build_type == 'Debug' }
-          expression { return params.iroha }
-        }
-      }
+    stage('Build') {
       parallel {
         stage ('x86_64_linux') {
           when {
             beforeAgent true
-            expression { return params.x86_64_linux }
+            anyOf {
+              expression { return params.x86_64_linux }
+              expression { return MERGE_CONDITIONS_SATISFIED == "true" }
+            }
           }
-          agent { label 'x86_64' }
+          agent { label 'x86_64_aws_build' }
           steps {
             script {
-              debugBuild = load ".jenkinsci/debug-build.groovy"
-              coverage = load ".jenkinsci/selected-branches-coverage.groovy"
-              if (coverage.selectedBranchesCoverage(['develop', 'master'])) {
-                debugBuild.doDebugBuild(true)
+              def debugBuild = load ".jenkinsci/debug-build.groovy"
+              def coverage = load ".jenkinsci/build-coverage.groovy"
+              if (params.build_type == 'Debug') {
+                debugBuild.doDebugBuild(coverage.checkCoverageConditions())
               }
-              else {
-                debugBuild.doDebugBuild()
-              }
-              if (GIT_LOCAL_BRANCH ==~ /(master|develop)/) {
-                releaseBuild = load ".jenkinsci/release-build.groovy"
+              if (params.build_type == 'Release') {
+                def releaseBuild = load ".jenkinsci/release-build.groovy"
                 releaseBuild.doReleaseBuild()
               }
             }
           }
           post {
-            always {
+            success {
               script {
-                post = load ".jenkinsci/linux-post-step.groovy"
-                post.linuxPostStep()
+                if (params.build_type == 'Release') {
+                  def post = load ".jenkinsci/post-step.groovy"
+                  post.postStep()
+                }
               }
             }
           }
         }
         stage('armv7_linux') {
           when {
             beforeAgent true
-            expression { return params.armv7_linux }
+            anyOf {
+              expression { return params.armv7_linux }
+              // expression { return MERGE_CONDITIONS_SATISFIED == "true" }
```

Comment:
What is the point of this line?

Reply:
It is disabled because current armv7 builds fail due to insufficient memory. Builds should be possible with cross-compilation, so this can serve as a reminder.